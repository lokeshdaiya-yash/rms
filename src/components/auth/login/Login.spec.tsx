import { BrowserRouter } from 'react-router-dom';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PtgUiLogin from './Login';
jest.mock('../services/auth.service.ts', () => {
  return {
    authClass: {
      login: () => Promise.resolve({}),
      getToken: () => JSON.stringify({ user: { username: 'tester' } }),
    },
  };
});
describe('PtgUiLogin', () => {
  let container;
  let getByTestId;
  let getByText;
  let getByLabelText;

  it('should create login component', () => {
    const {baseElement}= render(
      <BrowserRouter>
        <PtgUiLogin />
      </BrowserRouter>
    );
    expect(baseElement).toBeTruthy();
  });

  it(' Should Render login button', async () => {
    render(
      <BrowserRouter>
        <PtgUiLogin />
      </BrowserRouter>
    );
    const items = await screen.findAllByRole('button');
    expect(items).toHaveLength(1);
  });

  it('should render login form fields', async () => {
    const {container,getByTestId,getByText,getByLabelText} = render(
      <BrowserRouter>
        <PtgUiLogin />
      </BrowserRouter>
    );
    

    // text-inputs
    fireEvent.change(screen.getByTestId('email'), {
      target: { value: 'test@test' },
    });
    fireEvent.focus(screen.getByTestId('password'));
    fireEvent.change(screen.getByTestId('password'), {
      target: { value: 'test' },
    });
    fireEvent.change(screen.getByTestId('password'), {
      target: { value: '' },
    });
    fireEvent.focusOut(screen.getByTestId('password'));

    fireEvent.change(screen.getByTestId('email'), {
      target: { value: 'raj@gmail.com' },
    });

    fireEvent.change(screen.getByTestId('password'), {
      target: { value: '12345' },
    });

    // button
    fireEvent.click(screen.getByTestId('login'));

    userEvent.click(screen.getByTestId('login'));
     
  });
});
