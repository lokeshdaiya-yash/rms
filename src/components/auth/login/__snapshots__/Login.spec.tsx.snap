// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PtgUiLogin Should match Login Snapshot 1`] = `
<div
  className="login-wrapper container-fluid p-0 d-flex justify-content-center align-items-center"
>
  <div
    className="login-container"
  >
    <div
      className="login-form-wrapper"
    >
      <div
        className="form-group"
      />
      <div
        className="form-group"
      >
        <div
          className="login-logo text-center mb-3"
        >
          <img
            alt="IMG_LOGO"
            className=""
            src="./../../../assets/.png"
          />
        </div>
      </div>
      <form
        className="form-login"
      >
        <div
          className="login-form"
        >
          <div
            className="form-group mb-4 required"
          >
            <label
              htmlFor="inputEmail"
            >
              LABEL_EMAIL
            </label>
            <div
              className="userid input-group flex-nowrap"
            >
              <div
                className="input-group-prepend"
              >
                <i
                  className="login-user-icon"
                />
              </div>
              <input
                className="\\"w-100 form-control bg_0 "
                data-testid="email"
                disabled={false}
                id=""
                inputsize="lg"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="INPUT_PLACEHOLDER_EMAIL"
                required={true}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group mb-4 required"
          >
            <label
              htmlFor="inputPassword"
            >
              LABEL_PASSWORD
            </label>
            <div
              className="forgot-password float-end"
            >
              <div>
                <a
                  className="forgot-password float-end"
                  data-testid="linkForgotPassword"
                  onClick={[Function]}
                >
                  Forgot Password?
                </a>
              </div>
            </div>
            <div
              className="userid input-group flex-nowrap"
            >
              <div
                className="input-group-prepend"
              >
                <i
                  className="login-password-icon"
                />
              </div>
              <input
                className="\\"w-100 form-control bg_0 "
                data-testid="password"
                disabled={false}
                id=""
                inputsize="lg"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="INPUT_PLACEHOLDER_PASSWORD"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="new_label text-center mb-3"
        >
          <label>
            LABEL_INFO_MSG
             
            <a
              className="signup-btn"
              href="/signup"
              onClick={[Function]}
            >
              SIGN_UP
            </a>
            .
          </label>
        </div>
        <button
          className="w-100 btn btn-primary"
          data-testid="login"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          LOG_IN
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`PtgUiLogin changes the class when hovered 1`] = `
<div
  className="login-wrapper container-fluid p-0 d-flex justify-content-center align-items-center"
>
  <div
    className="login-container"
  >
    <div
      className="login-form-wrapper"
    >
      <div
        className="form-group"
      />
      <div
        className="form-group"
      >
        <div
          className="login-logo text-center mb-3"
        >
          <img
            alt="logo"
            className=""
            src="./../../../assets/logo.png"
          />
        </div>
      </div>
      <form
        className="form-login"
      >
        <div
          className="login-form"
        >
          <div
            className="form-group mb-4 required"
          >
            <label
              htmlFor="inputEmail"
            >
              Email
            </label>
            <div
              className="userid input-group flex-nowrap"
            >
              <div
                className="input-group-prepend"
              >
                <i
                  className="login-user-icon"
                />
              </div>
              <input
                className="\\"w-100 form-control bg_0 "
                data-testid="email"
                disabled={false}
                id=""
                inputsize="lg"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Enter your Email"
                required={true}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group mb-4 required"
          >
            <label
              htmlFor="inputPassword"
            >
              Password
            </label>
            <div
              className="forgot-password float-end"
            >
              <div>
                <a
                  className="forgot-password float-end"
                  data-testid="linkForgotPassword"
                  onClick={[Function]}
                >
                  Forgot Password?
                </a>
              </div>
            </div>
            <div
              className="userid input-group flex-nowrap"
            >
              <div
                className="input-group-prepend"
              >
                <i
                  className="login-password-icon"
                />
              </div>
              <input
                className="\\"w-100 form-control bg_0 "
                data-testid="password"
                disabled={false}
                id=""
                inputsize="lg"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Enter your password"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="new_label text-center mb-3"
        >
          <label>
            If you don't have an account click on 
             
            <a
              className="signup-btn"
              href="/signup"
              onClick={[Function]}
            >
              SIGN UP
            </a>
            .
          </label>
        </div>
        <button
          className="w-100 btn btn-primary"
          data-testid="login"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          LOG IN
        </button>
      </form>
    </div>
  </div>
</div>
`;
